app/models/sushi.rb

```diff
@@ -183,22 +187,38 @@ def coerce_metric_types(params = {}, allowed_types: ALLOWED_METRIC_TYPES)
   module QueryParameterValidation
     extend ActiveSupport::Concern
     included do
-      attr_reader :item_id, :item_id_in_params
-      attr_reader :platform, :platform_in_params
+      attr_reader :access_methods, :access_method_in_params, :item_id, :item_id_in_params, :platform, :platform_in_params
     end
 
+    ALLOWED_ACCESS_METHODS = ['regular'].freeze
+
+    ##
+    # @param params [Hash, ActionController::Parameters]
+    def validate_access_method(params)
+      allowed_access_methods_from_params = Array.wrap(params[:access_method].split('|')).map { |am| am.strip.downcase } & ALLOWED_ACCESS_METHODS
+
+      raise Sushi::InvalidParameterValue.invalid_access_method(params[:access_method], ALLOWED_ACCESS_METHODS) unless allowed_access_methods_from_params.any?
+
+      @access_methods = allowed_access_methods_from_params
+      @access_method_in_params = true
+    end
+
+    ##
+    # @param params [Hash, ActionController::Parameters]
     def validate_item_id(params)
       raise Sushi::NotFoundError.invalid_item_id(params[:item_id]) unless Hyrax::CounterMetric.exists?(work_id: params[:item_id])
 
       @item_id = params[:item_id]
-      @item_id_in_params = params.key?(:item_id)
+      @item_id_in_params = true
     end
 
+    ##
+    # @param params [Hash, ActionController::Parameters]
     def validate_platform(params, account)
-      raise Sushi::InvalidParameterValue.invalid_platform(params[:platform], account) unless params[:platform].blank? || params[:platform] == account.cname
```

Comment:
the `params[:platform].blank?` was written before we were checking that the param key exists before calling this method.

technically a user could write `platform=` and this exception wouldn't be raised, but I don't think that's a use case we have to account for.